utils/api_client.py

```
@@ -28,8 +34,27 @@ def request(self, method, path, **kwargs):
         response.raise_for_status()
         return response
 
-    def get(self, path, **kwargs):
-        return self.request('get', path, **kwargs).json()
+    def get(self, path, json=True, **kwargs):
+        response = self.request('get', path, **kwargs)
+        if response.status_code is 200:
```

I don't think it's possible to get this far unless the status code is ok. 
`response.raise_for_status()` in the request function should raise an exception before it gets here.
You can catch it to handle what to do if the response wasn't ok e.g.
```
try:
    response = self.request('get', path, **kwargs)
except requests.exceptions.HTTPError as e:
    # Do something here?
```

---

Ah I see what you mean.
I didn't notice that it's calling `response.raise_for_status()`
I don't think it should raise anything in the client as then the code has to use try/except but that's a diff conversation.

---

I'll leave it as is then we can revisit.